### ISSUE TITLE
Check 21 - Thresholds for Gather logs

### ISSUE THREAD

**Issue by JvanderMeer-GitHub** [ykuijs/SPPeriodicChecks#6]
I made some changes to the check to include a threshold, if the logs will contain x% warnings/errors it will report this as an issue.
Note that in the if section it doesn't check the warnings.

```PowerShell
$warningsThresholdPercentage  = 15
$errorsThresholdPercentage = 8
$topErrorsThreshold = 0


$ssas = Get-SPEnterpriseSearchServiceApplication

if ($null -eq $ssas)
{
    # No Search Service Applications found
    WriteLog "  Check not possible: No Search Service applications found. Either because none exist or an access issue occurred."
    $results.Check21 = $results.Check21 + "Search Gatherer Check: No Search Service applications found.`r`n"
    $results.Check21 = $results.Check21 + "`tEither because none exist or an access issue occurred.`r`n"
}
else
{
    # One or more Search Service Applications found
    foreach ($ssa in $ssas)
    {
        $logs = New-Object Microsoft.Office.Server.Search.Administration.CrawlLog $ssa

        $succeses = $logs.GetCrawledUrls($true,$logMaxRows,"",$false,$contentSourceID,0,$errorID,$startDate,$endDate)			
        $warnings = $logs.GetCrawledUrls($true,$logMaxRows,"",$false,$contentSourceID,1,$errorID,$startDate,$endDate)
        $errors = $logs.GetCrawledUrls($true,$logMaxRows,"",$false,$contentSourceID,2,$errorID,$startDate,$endDate)
        $topErrors = $logs.GetCrawledUrls($true,$logMaxRows,"",$false,$contentSourceID,4,$errorID,$startDate,$endDate)
		
		$warningsThreshold = [math]::Round(($succeses.documentcount * $warningsThresholdPercentage)/100)
		$errorsThreshold = [math]::Round(($succeses.documentcount * $errorsThresholdPercentage)/100)

        if (($errors.Rows[0]["DocumentCount"] -gt $errorsThreshold) -or `
            ($topErrors.Rows[0]["DocumentCount"] -gt $topErrorsThreshold))
        {
            WriteLog "  Check Failed"
            $results.Check21 = $results.Check21 + "Search Gatherer Check: $($ssa.name) - Failed`r`n"
            $results.Check21 = $results.Check21 + "`tWarnings: $($warnings.Rows[0]["DocumentCount"])`r`n"
            $results.Check21 = $results.Check21 + "`tErrors: $($errors.Rows[0]["DocumentCount"])`r`n"
            $results.Check21 = $results.Check21 + "`tTop Level Errors: $($topErrors.Rows[0]["DocumentCount"])`r`n"
        }
        else
        {
            WriteLog "  Check Passed"
            $results.Check21 = $results.Check21 + "Search Gatherer Check: $($ssa.name) - Passed`r`n"
        }
    }
}
```

**Comment by ykuijs** [comment]
Great suggestion. Will incorporate this into the script!

**Comment by ykuijs** [comment]
Added in next version

**Comment by ykuijs** [comment]
Fixed via PR #12
